src/main/java/org/jabref/logic/importer/fetcher/BiodiversityLibrary.java

```diff
@@ -0,0 +1,276 @@
+        uriBuilder.addParameter("id", identifier);
+
+        return uriBuilder.build().toURL();
+
+    }
+
+    public JSONObject getDetails(URL url) throws IOException {
+
+        URLDownload download = new URLDownload(url);
+        String response = download.asString();
+        return new JSONObject(response).getJSONArray("Result").getJSONObject(0);
+    }
+
+    public BibEntry getMostDetails(JSONObject item, BibEntry entry) throws IOException, URISyntaxException {
+
+        if (item.has("BHLType")) {
+            if (item.getString("BHLType").equals("Part")) {
+                URL url = getPartMetadataURL(item.getString("PartID"));
+                JSONObject itemsDetails = getDetails(url);
+                // Language
+                if (itemsDetails.has("Language")) {
+                    entry.setField(StandardField.LANGUAGE, itemsDetails.getString("Language"));
+                }
+                // DOI
+                if (itemsDetails.has("Doi")) {
+                    entry.setField(StandardField.DOI, itemsDetails.getString("Doi"));
+                }
+                // Publisher
+                if (itemsDetails.has("PublisherName")) {
+                    entry.setField(StandardField.PUBLISHER, itemsDetails.getString("PublisherName"));
+                }
+                // Volume
+                if (itemsDetails.has("Volume") && !entry.hasField(StandardField.VOLUME)) {
+                    entry.setField(StandardField.VOLUME, itemsDetails.getString("Volume"));
+                }
+                // Date
+                if (itemsDetails.has("Date") && !entry.hasField(StandardField.DATE) && !entry.hasField(StandardField.YEAR)) {
+                    entry.setField(StandardField.DATE, itemsDetails.getString("Date"));
+                }
+                // Link
+                if (itemsDetails.has("PartUrl")) {
+                    entry.setField(StandardField.URL, itemsDetails.getString("PartUrl"));
+                }
+
+            }
+
+            if (item.getString("BHLType").equals("Item")) {
+                URL url = getItemMetadataURL(item.getString("ItemID"));
+                JSONObject itemsDetails = getDetails(url);
+                // Editor
+                if (itemsDetails.has("Sponsor")) {
+                    entry.setField(StandardField.EDITOR, itemsDetails.getString("Sponsor"));
+                }
+                // Publisher
+                if (itemsDetails.has("HoldingInstitution")) {
+                    entry.setField(StandardField.PUBLISHER, itemsDetails.getString("HoldingInstitution"));
+                }
+                // Language
+                if (itemsDetails.has("Language")) {
+                    entry.setField(StandardField.LANGUAGE, itemsDetails.getString("Language"));
+                }
+                // Link
+                if (itemsDetails.has("ItemUrl")) {
+                    entry.setField(StandardField.URL, itemsDetails.getString("ItemUrl"));
+                }
```

Comment:
When not using logging, you can shorten it to:

```suggestion
                entry.setField(StandardField.URL, itemsDetails.optString("ItemUrl", ""));                
```